middleware/backend/setup.go

```diff
@@ -0,0 +1,24 @@
+package backend
+
+import (
+	"github.com/coredns/coredns/middleware"
+	"github.com/coredns/coredns/middleware/backend/msg"
+	"github.com/coredns/coredns/middleware/proxy"
+)
+
+// StubServiceBackend is a ServiceBackend that also support stubs.
+type StubServiceBackend interface {
+	middleware.ServiceBackend
+
+	Records(name string, exact bool) ([]msg.Service, error)
```

Comment:
Why the `Services` method in `middleware.ServiceBackend` not sufficient, why this extra method?